dataflow/snippets/src/main/java/com/example/dataflow/BigQueryReadAvro.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dataflow;
+
+// [START dataflow_bigquery_read_avro]
+import org.apache.avro.util.Utf8;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.coders.DefaultCoder;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.TypedRead;
+import org.apache.beam.sdk.io.gcp.bigquery.SchemaAndRecord;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+public class BigQueryReadAvro {
+
+  // A custom datatype to hold a record from the source table.
+  @DefaultCoder(AvroCoder.class)
+  public static class MyData {
+    public String name;
+    public Long age;
+
+    // Function to convert Avro records to MyData instances.
+    public static class FromSchemaAndRecord
+            implements SerializableFunction<SchemaAndRecord, MyData> {
+      @Override public MyData apply(SchemaAndRecord elem) {
+        var data = new MyData();
+        var record = elem.getRecord();
+        data.name = ((Utf8) record.get("user_name")).toString();
+        data.age = (Long) record.get("age");
+        return data;
```

Comment:
@averikitsch  Do we have a precedence for allowing the use of `var` in samples? 

Since they are samples, I think we used the full type so readers can easily understand the returns and what each line is doing.

Reply:
Updated to replace 'var' throughout